pyuvdata/utils.py

```diff
@@ -1059,9 +1061,1341 @@ def unphase_uvw(ra, dec, uvw):
+                )
+    coord_object = SkyCoord(
+        (np.repeat(lon_coord, len(time_array)) if rep_crds else lon_coord) * units.rad,
+        (np.repeat(lat_coord, len(time_array)) if rep_crds else lat_coord) * units.rad,
+        frame=in_coord_frame,
+        equinox=in_coord_epoch
+        if (isinstance(in_coord_epoch, str) or isinstance(in_coord_epoch, Time))
+        else (None if in_coord_epoch is None else "J%f" % in_coord_epoch),
+        obstime=None
+        if time_array is None
+        else Time(
+            np.repeat(time_array, len(lon_coord)) if rep_time else time_array,
+            scale="utc",
+            format="jd",
+        ),
+    )
+
+    new_coord = coord_object.transform_to(out_coord_frame)
+    return new_coord.spherical.lon.rad, new_coord.spherical.lat.rad
+
+
+def translate_icrs_to_app(
+    time_array,
+    ra_coord,
+    dec_coord,
+    telescope_lat,
+    telescope_lon,
+    telescope_alt,
+    pm_ra=None,
+    pm_dec=None,
+    rad_vel=None,
+    parallax=None,
+    use_novas=False,
+):
+    """
+    Translate a set of coordinates in ICRS to topocentric/apparent coordinates.
+
+    This utility uses either astropy or NOVAS to calculate the apparent (i.e.,
+    topocentric) coordinates of a source at a given time and location, given
+    a set of coordinates expressed in the ICRS frame. These coordinates are most
+    typically used for defining the phase center of the array (i.e, calculating
+    baseline vectors).
+
+    As of astropy v4.2, the agreement between astropy and NOVAS is consistent
+    down to the level of approximately 1 milliarcsec. This is done through a
+    slight modification to the TETE observing frame, by applying polar motion
+    vectors from IERS to the aforementioned frame.
+
+    Parameters
+    ----------
+    time_array : float or ndarray of float
+        Julian dates to calculate coordinate positions for. Can either be a single
+        float, or an ndarray of shape is (Ntimes,).
+    ra_coord : float or ndarray of float
+        ICRS RA of the celestial target, expressed in units of radians. Can either
+        be a single float or array of shape (Ntimes,), although this must be consistent
+        with other parameters (with the exception of telescope location parameters).
+    dec_coord : float or ndarray of float
+        ICRS Dec of the celestial target, expressed in units of radians. Can either
+        be a single float or array of shape (Ntimes,), although this must be consistent
+        with other parameters (with the exception of telescope location parameters).
+    telescope_lon : float
+        ITRF longitude of the phase center of the array, expressed in units of radians.
+    telescope_lat : float
+        ITRF latitude of the phase center of the array, expressed in units of radians.
+    telescope_alt : float
+        Altitude (rel to sea-level) of the phase center of the array, expressed in
+        units of meters.
```

Comment:
These arguments to this function (`translate_icrs_to_app`) are ordered as latitude, longitude, altitude so it would help reduce the potential for confusion to document them in the same order.

Reply:
I actually decided to do away with the separate lat, lon, alt arguments in favor of a single argument that's normally a tuple containing all three, since 1) it makes it more readily compatible w/ `UVData` objects, and 2) it also allows me to pass around `EarthLocation` objects, which were being separately constructed within each function.